policies/lib/tree/30_generic_methods/file_replace_lines.cf

```diff
@@ -55,44 +55,31 @@
 bundle agent file_replace_lines(path, line, replacement)
 {
   vars:
-    pass1::
-      "old_class_prefix"  string => canonify("file_replace_lines_${path}");
-
-    # If replacement contains ${match.x}, it would not evaluate to anything outside of file promises, so we
-    # need to exclude it from class prefix
-    pass1.replacement_non_eval::
-      "args"               slist => { "${path}", "${line}" };
-
-    pass1.!replacement_non_eval::
-      "args"               slist => { "${path}", "${line}", "${replacement}" };
-
-    pass1::
-      "report_param"      string => join("_", args);
-      "full_class_prefix" string => canonify("file_replace_lines_${report_param}");
-      "class_prefix"      string => string_head("${full_class_prefix}", "1000");
+      "class_prefix"       string => canonify("file_replace_lines_${path}");
+      "inner_class_prefix" string => "${report_data.method_id}_inner";
 
   classes:
-      "replacement_defined"           not => strcmp("${replacement}", "");
-      "no_replacement_defined" expression => strcmp("${replacement}", "");
-
-      # if both classes are undefined then there is most likely a "match.x" in it, that is not evaluated
-      # by the agent (see https://issues.rudder.io/issues/15029 and https://issues.rudder.io/issues/14286
-      "replacement_non_eval"   expression => "!replacement_defined.!no_replacement_defined";
```

Comment:
is this problem fixed now?

Reply:
King of, it seems that it was only a reporting issue as the args variable could contain undefined variables calls (the ${match.x}). Now that we are using uuid and automatic resulting classes, it should not be an issue.